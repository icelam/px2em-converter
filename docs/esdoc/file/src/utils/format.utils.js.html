<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  <base data-ice="baseUrl" href="../../../">
  <title data-ice="title">src/utils/format.utils.js | PX2Em Converter</title>
  <link type="text/css" rel="stylesheet" href="css/style.css">
  <link type="text/css" rel="stylesheet" href="css/prettify-tomorrow.css">
  <script src="script/prettify/prettify.js"></script>
  <script src="script/manual.js"></script>
<meta name="description" content="PX2EM is a conversion tool which allows you to calculate the em sizes from pixels. Just add the base font size and the pixels to convert, and you&apos;re done!"><meta property="twitter:card" content="summary"><meta property="twitter:title" content="PX2Em Converter"><meta property="twitter:description" content="PX2EM is a conversion tool which allows you to calculate the em sizes from pixels. Just add the base font size and the pixels to convert, and you&apos;re done!"><meta property="twitter:image" content="./preview.png"></head>
<body class="layout-container" data-ice="rootContainer">

<header>
  <a href="./" style="display: flex; align-items: center;"><img src="./image/brand_logo.png" style="width:34px;"></a>
  
  <a href="identifiers.html">Reference</a>
  <a href="source.html">Source</a>
  <a href="test.html" data-ice="testLink">Test</a>
  <div class="search-box">
  <span>
    <img src="./image/search.png">
    <span class="search-input-edge"></span><input class="search-input"><span class="search-input-edge"></span>
  </span>
    <ul class="search-result"></ul>
  </div>
<a href="http://github.com/icelam/px2em-converter">Repository</a></header>

<nav class="navigation" data-ice="nav"><div>
  <ul>
    
  <li data-ice="doc"><a data-ice="dirPath" class="nav-dir-path" href="identifiers.html#hooks">hooks</a><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-useDebounce">useDebounce</a></span></span></li>
<li data-ice="doc"><a data-ice="dirPath" class="nav-dir-path" href="identifiers.html#utils">utils</a><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-convertHexToRgb">convertHexToRgb</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-convertPixelRangeToArray">convertPixelRangeToArray</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-deduplicateArray">deduplicateArray</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-removeExtraSpacesAndCommas">removeExtraSpacesAndCommas</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-roundToNearest">roundToNearest</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-sortArrayAscending">sortArrayAscending</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-clearAll">clearAll</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-getKey">getKey</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-removeKey">removeKey</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-setKey">setKey</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-isValidFontSize">isValidFontSize</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-isValidPositiveInteger">isValidPositiveInteger</a></span></span></li>
</ul>
</div>
</nav>

<div class="content" data-ice="content"><h1 data-ice="title">src/utils/format.utils.js</h1>
<pre class="source-code line-number raw-source-code"><code class="prettyprint linenums" data-ice="content">import { isValidPositiveInteger } from &apos;./validate.utils&apos;;

/**
 * Coverts any HEX color code to RGB
 * @param {number} value - Interger / floating point value that needs to be rounded
 * @param {number} precision - Number of decimal places
 * @returns {number} Rounded floating point value
 */
export const roundToNearest = (value, precision) =&gt; {
  if (!Number.isInteger(precision)) {
    return value;
  }
  // parseFloat for trimming trailing zeros
  return parseFloat(value.toFixed(precision));
};

/**
 * Deduplicates any array
 * @param {array} arr - Array to be deduplicated
 * @returns {array} Deduplicated array If `arr` provided is not an array, returns an empty array
 */
export const deduplicateArray = (arr) =&gt; {
  if (!Array.isArray(arr)) {
    return [];
  }

  const arraySet = new Set(arr);
  return Array.from(arraySet);
};

/**
 * Remove multiple consecutive spaces, trialing space and comma
 * @param {string} str - String to process
 * @returns {string} String which its multiple consecutive spaces,
 * trialing space and comma are removed
 */
export const removeExtraSpacesAndCommas = (str) =&gt; str
  .replace(/\s+/g, &apos; &apos;)
  .replace(/\s$/, &apos;&apos;)
  .replace(/,$/, &apos;&apos;);

/**
 * Sort array in ascending order
 * @param {array} arr - Array to sort
 * @returns {array} Array sorted in ascending order
 */
const sortArrayAscending = (arr) =&gt; {
  const newArray = [...arr];
  return newArray.sort((a, b) =&gt; a - b);
};

/**
 * Convert pixel range to list of pixels
 * @param {string} str - Pixel range, e.g. `&quot;12 - 20, 50, 100&quot;`
 * @returns {array} List of pixels, e.g. `[12, 13, 14, 15, 16, 17, 18, 19, 20, 50, 100]``
 */
export const convertPixelRangeToArray = (str) =&gt; {
  try {
    const pixels = [];

    if (!str) {
      throw new Error(&apos;EMPTY_INPUT&apos;);
    }

    const formattedString = removeExtraSpacesAndCommas(str);
    const rangeItems = formattedString.split(&apos;,&apos;);

    rangeItems.forEach((item) =&gt; {
      let currentDigit = &apos;&apos;;
      const operations = [];

      const pushDigit = () =&gt; {
        operations.push(currentDigit);
        currentDigit = &apos;&apos;;
      };

      /**
       * Parse string to operations:
       *  - &quot;11&quot; to [&quot;11&quot;]
       *  - &quot;11 - 13&quot; to [&quot;11&quot;, &quot;-&quot;, &quot;13&quot;]
       *  - &quot;1-3&quot; to [&quot;1&quot;, &quot;-&quot;, &quot;1&quot;]
       */
      for (let i = 0; i &lt; item.length; i++) {
        if (item[i] === &apos; &apos;) {
          currentDigit &amp;&amp; pushDigit();
        } else if (isValidPositiveInteger(item[i])) {
          currentDigit += item[i];
        } else if (item[i] === &apos;-&apos;) {
          currentDigit &amp;&amp; pushDigit();
          operations.push(&apos;-&apos;);
        } else {
          throw new Error(&apos;INVALID_INPUT&apos;);
        }
      }

      currentDigit &amp;&amp; pushDigit();

      /**
       * Convert operations to pixels:
       *  - Valid Case 1: Single positive integer
       *  - Valid Case 2: Pixel range (e.g. 1-10)
       */
      if (operations.length !== 1 &amp;&amp; operations.length !== 3) {
        throw new Error(&apos;INVALID_OPERATION_LENGTH&apos;);
      }

      if (operations.length === 1) {
        if (isValidPositiveInteger(operations[0])) {
          pixels.push(parseInt(operations[0], 10));
        } else {
          throw new Error(&apos;INVALID_NUMBER&apos;);
        }
      } else {
        const [leftString, operator, rightString] = operations;
        if (isValidPositiveInteger(leftString) &amp;&amp; operator === &apos;-&apos; &amp;&amp; isValidPositiveInteger(rightString)) {
          const leftNumber = parseInt(leftString, 10);
          const rightNumber = parseInt(rightString, 10);
          if (leftNumber &lt;= rightNumber) {
            for (let i = leftNumber; i &lt;= rightNumber; i++) {
              pixels.push(parseInt(i, 10));
            }
          } else {
            throw new Error(&apos;INVALID_RANGE&apos;);
          }
        } else {
          throw new Error(&apos;INVALID_OPERATION&apos;);
        }
      }
    });

    const uniquePixels = deduplicateArray(pixels);
    const sortedPixels = sortArrayAscending(uniquePixels);
    return sortedPixels;
  } catch (error) {
    return [];
  }
};
</code></pre>

</div>

<footer class="footer">
  Generated by <a href="https://esdoc.org">ESDoc<span data-ice="esdocVersion">(1.1.0)</span><img src="./image/esdoc-logo-mini-black.png"></a>
</footer>

<script src="script/search_index.js"></script>
<script src="script/search.js"></script>
<script src="script/pretty-print.js"></script>
<script src="script/inherited-summary.js"></script>
<script src="script/test-summary.js"></script>
<script src="script/inner-link.js"></script>
<script src="script/patch-for-local.js"></script>
</body>
</html>
