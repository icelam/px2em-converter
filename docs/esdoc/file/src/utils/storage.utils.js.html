<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  <base data-ice="baseUrl" href="../../../">
  <title data-ice="title">src/utils/storage.utils.js | PX2EM Converter</title>
  <link type="text/css" rel="stylesheet" href="css/style.css">
  <link type="text/css" rel="stylesheet" href="css/prettify-tomorrow.css">
  <script src="script/prettify/prettify.js"></script>
  <script src="script/manual.js"></script>
<meta name="description" content="PX2EM is a conversion tool which allows you to calculate the em sizes from pixels. Just add the base font size and the pixels to convert, and you&apos;re done!"><meta property="twitter:card" content="summary"><meta property="twitter:title" content="PX2EM Converter"><meta property="twitter:description" content="PX2EM is a conversion tool which allows you to calculate the em sizes from pixels. Just add the base font size and the pixels to convert, and you&apos;re done!"><meta property="twitter:image" content="./preview.png"></head>
<body class="layout-container" data-ice="rootContainer">

<header>
  <a href="./" style="display: flex; align-items: center;"><img src="./image/brand_logo.png" style="width:34px;"></a>
  
  <a href="identifiers.html">Reference</a>
  <a href="source.html">Source</a>
  <a href="test.html" data-ice="testLink">Test</a>
  <div class="search-box">
  <span>
    <img src="./image/search.png">
    <span class="search-input-edge"></span><input class="search-input"><span class="search-input-edge"></span>
  </span>
    <ul class="search-result"></ul>
  </div>
<a href="http://github.com/icelam/px2em-converter">Repository</a></header>

<nav class="navigation" data-ice="nav"><div>
  <ul>
    
  <li data-ice="doc"><a data-ice="dirPath" class="nav-dir-path" href="identifiers.html#hooks">hooks</a><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-useDebounce">useDebounce</a></span></span></li>
<li data-ice="doc"><a data-ice="dirPath" class="nav-dir-path" href="identifiers.html#utils">utils</a><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-convertHexToRgb">convertHexToRgb</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-convertPixelRangeToArray">convertPixelRangeToArray</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-deduplicateArray">deduplicateArray</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-removeExtraSpacesAndCommas">removeExtraSpacesAndCommas</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-roundToNearest">roundToNearest</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-sortArrayAscending">sortArrayAscending</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-clearAll">clearAll</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-getKey">getKey</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-removeKey">removeKey</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-setKey">setKey</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-isValidFontSize">isValidFontSize</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-isValidPositiveInteger">isValidPositiveInteger</a></span></span></li>
</ul>
</div>
</nav>

<div class="content" data-ice="content"><h1 data-ice="title">src/utils/storage.utils.js</h1>
<pre class="source-code line-number raw-source-code"><code class="prettyprint linenums" data-ice="content">/**
 * Store data to local storage when key is provided
 * @param {string} key - Local storage key which specifies where the data to be stored
 * @param {*} data - Data to store to local storage
 */
const setKey = (key, data) =&gt; {
  if (!key) { return; }
  window.localStorage.setItem(key, data);
};

/**
 * Get the data stored in specific key of local storage
 * @property {function()} localStorageUtils - Helpers for interacting with local storage
 * @param {string} key - Local storage key which specifies where the data to be stored
 * @returns {string} Data stored in `key` of local storage
 */
const getKey = (key) =&gt; window.localStorage.getItem(key);

/**
 * Remove the data stored in specific key of local storage
 * @param {string} key - Local storage key which specifies where the data to be stored
 */
const removeKey = (key) =&gt; {
  window.localStorage.removeItem(key);
};

/**
 * Clear all data stored in local storage
 */
const clearAll = () =&gt; {
  window.localStorage.clear();
};

export default {
  setKey,
  getKey,
  removeKey,
  clearAll
};
</code></pre>

</div>

<footer class="footer">
  Generated by <a href="https://esdoc.org">ESDoc<span data-ice="esdocVersion">(1.1.0)</span><img src="./image/esdoc-logo-mini-black.png"></a>
</footer>

<script src="script/search_index.js"></script>
<script src="script/search.js"></script>
<script src="script/pretty-print.js"></script>
<script src="script/inherited-summary.js"></script>
<script src="script/test-summary.js"></script>
<script src="script/inner-link.js"></script>
<script src="script/patch-for-local.js"></script>
</body>
</html>
