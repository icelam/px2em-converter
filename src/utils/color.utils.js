const convertHexToRgb = (color) => {
  const colorCode = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(color.toLowerCase());

  return colorCode
    ? `${parseInt(colorCode[1], 16)}, ${parseInt(colorCode[2], 16)}, ${parseInt(colorCode[3], 16)}`
    : '';
};

export default convertHexToRgb;
