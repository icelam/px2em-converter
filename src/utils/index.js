export { default as convertHexToRgb } from './color.utils';
export * from './format.utils';
export { default as localStorageUtils } from './storage.utils';
export * from './validate.utils';
