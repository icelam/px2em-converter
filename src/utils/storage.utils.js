const setKey = (key, data) => {
  if (!key) { return; }
  window.localStorage.setItem(key, data);
};

const getKey = (key) => window.localStorage.getItem(key);

const removeKey = (key) => {
  window.localStorage.removeItem(key);
};

const clearAll = () => {
  window.localStorage.clear();
};

export default {
  setKey,
  getKey,
  removeKey,
  clearAll
};
