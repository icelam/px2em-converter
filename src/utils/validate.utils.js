/**
 * Validate if user input is a positive integer
 * @param {string} str - User input
 * @returns {boolean} `true` for positive integers, `false` for negative integers and other inputs
 */
export const isValidPositiveInteger = (str) => {
  const re = /^\d+$/;
  return re.test(str);
};

/**
 * Validate if user input is a valid font size
 * @param {string} str - User input
 * @returns {boolean} `true` for decimal and integer values, `false` for other inputs
 */
export const isValidFontSize = (str) => {
  const re = /^[0-9]+(\.[0-9]+)?$/;
  return re.test(str);
};
