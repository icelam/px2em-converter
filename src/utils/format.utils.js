import { isValidPositiveInteger } from './validate.utils';

export const roundToNearest = (value, precision) => {
  if (!Number.isInteger(precision)) {
    return value;
  }
  // parseFloat for trimming trailing zeros
  return parseFloat(value.toFixed(precision));
};

export const deduplicateArray = (arr) => {
  if (!Array.isArray(arr)) {
    return [];
  }

  const arraySet = new Set(arr);
  return Array.from(arraySet);
};

export const removeExtraSpacesAndCommas = (str) => str
  .replace(/\s+/g, ' ')
  .replace(/\s$/, '')
  .replace(/,$/, '');

const sortArrayAscending = (arr) => {
  const newArray = [...arr];
  return newArray.sort((a, b) => a - b);
};

export const convertPixelRangeToArray = (str) => {
  try {
    const pixels = [];

    if (!str) {
      throw new Error('EMPTY_INPUT');
    }

    const formattedString = removeExtraSpacesAndCommas(str);
    const rangeItems = formattedString.split(',');

    rangeItems.forEach((item) => {
      let currentDigit = '';
      const operations = [];

      const pushDigit = () => {
        operations.push(currentDigit);
        currentDigit = '';
      };

      /**
       * Parse string to operations:
       *  - "11" to ["11"]
       *  - "11 - 13" to ["11", "-", "13"]
       *  - "1-3" to ["1", "-", "1"]
       */
      for (let i = 0; i < item.length; i++) {
        if (item[i] === ' ') {
          currentDigit && pushDigit();
        } else if (isValidPositiveInteger(item[i])) {
          currentDigit += item[i];
        } else if (item[i] === '-') {
          currentDigit && pushDigit();
          operations.push('-');
        } else {
          throw new Error('INVALID_INPUT');
        }
      }

      currentDigit && pushDigit();

      /**
       * Convert operations to pixels:
       *  - Valid Case 1: Single positive integer
       *  - Valid Case 2: Pixel range (e.g. 1-10)
       */
      if (operations.length !== 1 && operations.length !== 3) {
        throw new Error('INVALID_OPERATION_LENGTH');
      }

      if (operations.length === 1) {
        if (isValidPositiveInteger(operations[0])) {
          pixels.push(parseInt(operations[0], 10));
        } else {
          throw new Error('INVALID_NUMBER');
        }
      } else {
        const [leftString, operator, rightString] = operations;
        if (isValidPositiveInteger(leftString) && operator === '-' && isValidPositiveInteger(rightString)) {
          const leftNumber = parseInt(leftString, 10);
          const rightNumber = parseInt(rightString, 10);
          if (leftNumber <= rightNumber) {
            for (let i = leftNumber; i <= rightNumber; i++) {
              pixels.push(parseInt(i, 10));
            }
          } else {
            throw new Error('INVALID_RANGE');
          }
        } else {
          throw new Error('INVALID_OPERATION');
        }
      }
    });

    const uniquePixels = deduplicateArray(pixels);
    const sortedPixels = sortArrayAscending(uniquePixels);
    return sortedPixels;
  } catch (error) {
    return [];
  }
};
