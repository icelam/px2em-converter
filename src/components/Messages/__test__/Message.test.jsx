import React from 'react';
import { mountWithProvider } from '../../../../jest.utils';
import message, { Message } from '..';

const successMessage = 'Success Message!';
const warnMessage = 'Warning Message!';
const infoMessage = 'Info Message!';
const errorMessage = 'Error Message!';
const loadingMessage = 'Loading Message!';

describe('Icon', () => {
  it('should render', () => {
    const wrapper = mountWithProvider(<Message type="success" message={successMessage} />);
    expect(wrapper.find('span.message')).toHaveLength(1);
    expect(wrapper.find('span.message i svg')).toHaveLength(1);
  });

  it('should display success message', () => {
    const wrapper = mountWithProvider(<Message type="success" message={successMessage} />);
    expect(wrapper.find('span.message')).toHaveLength(1);
    expect(wrapper.find('span.message i svg')).toHaveLength(1);
    expect(wrapper.find('span.message i svg').text()).toEqual('success.svg');
    expect(wrapper.find('span.message span').text()).toEqual(successMessage);
    expect(wrapper).toMatchSnapshot();
  });

  it('should display warning message', () => {
    const wrapper = mountWithProvider(<Message type="warn" message={warnMessage} />);
    expect(wrapper.find('span.message')).toHaveLength(1);
    expect(wrapper.find('span.message i svg')).toHaveLength(1);
    expect(wrapper.find('span.message i svg').text()).toEqual('warn.svg');
    expect(wrapper.find('span.message span').text()).toEqual(warnMessage);
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('should display info message', () => {
    const wrapper = mountWithProvider(<Message type="info" message={infoMessage} />);
    expect(wrapper.find('span.message')).toHaveLength(1);
    expect(wrapper.find('span.message i svg')).toHaveLength(1);
    expect(wrapper.find('span.message i svg').text()).toEqual('info.svg');
    expect(wrapper.find('span.message span').text()).toEqual(infoMessage);
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('should display error message', () => {
    const wrapper = mountWithProvider(<Message type="error" message={errorMessage} />);
    expect(wrapper.find('span.message')).toHaveLength(1);
    expect(wrapper.find('span.message i svg')).toHaveLength(1);
    expect(wrapper.find('span.message i svg').text()).toEqual('error.svg');
    expect(wrapper.find('span.message span').text()).toEqual(errorMessage);
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('should display loading message', () => {
    const wrapper = mountWithProvider(<Message type="loading" message={loadingMessage} />);
    expect(wrapper.find('span.message')).toHaveLength(1);
    expect(wrapper.find('span.message i svg')).toHaveLength(1);
    expect(wrapper.find('span.message i svg').text()).toEqual('loading.svg');
    expect(wrapper.find('span.message span').text()).toEqual(loadingMessage);
    expect(wrapper.render()).toMatchSnapshot();
  });
});

describe(('message'), () => {
  it('should render', () => {
    message.success(successMessage);
    expect(document.querySelectorAll('.message-notification-notice')).toHaveLength(1);
  });

  it('should display 3 message at most', () => {
    message.success(successMessage, 10);
    message.success(successMessage, 10);
    message.success(successMessage, 10);
    message.success(successMessage, 10);
    expect(document.querySelectorAll('.message-notification-notice')).toHaveLength(3);
  });

  it('should trigger success message when calling .success()', () => {
    message.success(successMessage);
    expect(document.querySelectorAll('.message-notification-notice')).toHaveLength(3);
    expect(document.querySelector('.message-notification-notice:nth-child(3)').textContent).toContain(successMessage);
  });

  it('should trigger warning message when calling .warning()', () => {
    message.warn(warnMessage);
    expect(document.querySelectorAll('.message-notification-notice')).toHaveLength(3);
    expect(document.querySelector('.message-notification-notice:nth-child(3)').textContent).toContain(warnMessage);
  });

  it('should trigger info message when calling .info()', () => {
    message.info(infoMessage);
    expect(document.querySelectorAll('.message-notification-notice')).toHaveLength(3);
    expect(document.querySelector('.message-notification-notice:nth-child(3)').textContent).toContain(infoMessage);
  });

  it('should trigger error message when calling .error()', () => {
    message.error(errorMessage);
    expect(document.querySelectorAll('.message-notification-notice')).toHaveLength(3);
    expect(document.querySelector('.message-notification-notice:nth-child(3)').textContent).toContain(errorMessage);
  });

  it('should trigger loading message when calling .loading()', () => {
    message.loading(loadingMessage);
    expect(document.querySelectorAll('.message-notification-notice')).toHaveLength(3);
    expect(document.querySelector('.message-notification-notice:nth-child(3)').textContent).toContain(loadingMessage);
  });
});
