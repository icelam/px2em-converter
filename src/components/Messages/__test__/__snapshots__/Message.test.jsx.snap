// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should display error message 1`] = `
.c1 {
  line-height: 1em;
  vertical-align: middle;
}

.c1 > svg {
  height: 1em;
  fill: #ffffff;
}

.c0 {
  background-color: #828b8e;
  color: #ffffff;
  padding: 0.5rem;
  font-size: 0.875rem;
  line-height: 1em;
  display: inline-block;
  margin: 0.25rem;
  box-shadow: 0 0 0.3125rem rgba(0,0,0,0.7);
}

<span
  class="c0 message"
>
  <i
    class="c1 svg-icon"
  >
    <svg>
      error.svg
    </svg>
  </i>
   
  <span>
    Error Message!
  </span>
</span>
`;

exports[`Icon should display info message 1`] = `
.c1 {
  line-height: 1em;
  vertical-align: middle;
}

.c1 > svg {
  height: 1em;
  fill: #ffffff;
}

.c0 {
  background-color: #828b8e;
  color: #ffffff;
  padding: 0.5rem;
  font-size: 0.875rem;
  line-height: 1em;
  display: inline-block;
  margin: 0.25rem;
  box-shadow: 0 0 0.3125rem rgba(0,0,0,0.7);
}

<span
  class="c0 message"
>
  <i
    class="c1 svg-icon"
  >
    <svg>
      info.svg
    </svg>
  </i>
   
  <span>
    Info Message!
  </span>
</span>
`;

exports[`Icon should display loading message 1`] = `
.c1 {
  line-height: 1em;
  vertical-align: middle;
}

.c1 > svg {
  height: 1em;
  fill: #ffffff;
}

.c2 {
  -webkit-animation: cqkDSr 1s infinite linear;
  animation: cqkDSr 1s infinite linear;
}

.c0 {
  background-color: #828b8e;
  color: #ffffff;
  padding: 0.5rem;
  font-size: 0.875rem;
  line-height: 1em;
  display: inline-block;
  margin: 0.25rem;
  box-shadow: 0 0 0.3125rem rgba(0,0,0,0.7);
}

<span
  class="c0 message"
>
  <i
    class="c1 svg-icon"
  >
    <svg
      class="c2"
    >
      loading.svg
    </svg>
  </i>
   
  <span>
    Loading Message!
  </span>
</span>
`;

exports[`Icon should display success message 1`] = `
.c1 {
  line-height: 1em;
  vertical-align: middle;
}

.c1 > svg {
  height: 1em;
  fill: #ffffff;
}

.c0 {
  background-color: #828b8e;
  color: #ffffff;
  padding: 0.5rem;
  font-size: 0.875rem;
  line-height: 1em;
  display: inline-block;
  margin: 0.25rem;
  box-shadow: 0 0 0.3125rem rgba(0,0,0,0.7);
}

<Message
  message="Success Message!"
  type="success"
>
  <styled.span
    className="message"
  >
    <span
      className="c0 message"
    >
      <Icon
        type="success"
      >
        <styled.i
          className="svg-icon"
        >
          <i
            className="c1 svg-icon"
          >
            <ForwardRef(SvgSuccess)>
              <svg>
                success.svg
              </svg>
            </ForwardRef(SvgSuccess)>
          </i>
        </styled.i>
      </Icon>
       
      <span>
        Success Message!
      </span>
    </span>
  </styled.span>
</Message>
`;

exports[`Icon should display warning message 1`] = `
.c1 {
  line-height: 1em;
  vertical-align: middle;
}

.c1 > svg {
  height: 1em;
  fill: #ffffff;
}

.c0 {
  background-color: #828b8e;
  color: #ffffff;
  padding: 0.5rem;
  font-size: 0.875rem;
  line-height: 1em;
  display: inline-block;
  margin: 0.25rem;
  box-shadow: 0 0 0.3125rem rgba(0,0,0,0.7);
}

<span
  class="c0 message"
>
  <i
    class="c1 svg-icon"
  >
    <svg>
      warn.svg
    </svg>
  </i>
   
  <span>
    Warning Message!
  </span>
</span>
`;
